dpnp/dpnp_iface_mathematical.py

```diff
@@ -2742,6 +2778,181 @@ def gradient(f, *varargs, axis=None, edge_order=1):
 )
 
 
+def interp(x, xp, fp, left=None, right=None, period=None):
+    """
+    One-dimensional linear interpolation.
+
+    Returns the one-dimensional piecewise linear interpolant to a function
+    with given discrete data points (`xp`, `fp`), evaluated at `x`.
+
+    For full documentation refer to :obj:`numpy.interp`.
+
+    Parameters
+    ----------
+    x : {dpnp.ndarray, usm_ndarray}
```

Comment:
Do we need to state that both `x` and `xp` are expected to have a real-valued floating-point data type (see other ufuncs for a reference)?

Reply:
done